woocommerce/woocommerce-gutenberg-products-block
# Inject block attributes rather than save them

Currently, we pass saved attributes to our frontend code using HTML `data-` attributes which get added to the saved block markup via `save`. 

See: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/blocks/cart-checkout/cart/index.js#L46-L67

The big problem with this is that when block attributes change, get added or removed, or deprecated, we either **cause block invalidation errors** or need to write lengthly backwards compatibility code to handle the migration. 

This PR contains an alternative solution; rather than save attributes to the block, it _dynamically injects attributes via the PHP render function_. That means our frontend blocks still have access to the same `data-` attributes, we just add them during render automagically.

Closes #1997 

**Note, this invalidates current cart/checkout blocks. I have no implemented this for the All Products block which is in production.**

### How to test the changes in this Pull Request:

1. Create cart/checkout blocks.
2. Set attributes.
3. Check attributes persist on frontend.
